### PR TITLE
Select: Fixes virtualized select showing empty space above selected value 

### PR DESCRIPTION
**Pull request by torkelo**
Story that replicates the https://github.com/grafana/grafana/issues/84542 issue 

https://github.com/grafana/grafana/assets/10999/e7556bf2-9b86-40f0-a2ca-53d4ffaa6d6d

Looks like it's caused by initialScrollOffset  here: 
https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Select/SelectMenu.tsx#L71

I tried upgrading react-window in case it was a bug in it but that did not fix it. 

I ended up adding some limits / logic to intitalScrollOffset trying to optimize it to scroll so item is in the middle of the menu. Fixes issue and improves the scroll to value logic

### CODE REVIEW

File: packages/grafana-ui/src/components/Select/Select.story.tsx [hunk 1]
```diff
@@ -123,6 +124,30 @@ export const BasicVirtualizedList: Story<StoryProps> = (args) => {
     </>
   );
 };
+
+export const BasicVirtualizedListBug: Story<StoryProps> = (args) => {
```
[the remark this answers is not in the document]
Comment (torkelo) [reply packages/grafana-ui/src/components/Select/Select.story.tsx:128]: @Clarity-89 not can name it differently. The bug only shows up when you have few items in a virtualized list

[the remark this answers is not in the document]
Comment (Clarity-89) [reply packages/grafana-ui/src/components/Select/Select.story.tsx]: Ah ok, can also keep it and name `LongVirtualizedList` or smth like that.

**Review by Clarity-89 — APPROVED** [review]
Tested and this seems to fix the issue 🎉

**Review by eledobleefe — APPROVED** [review]
Tested on local and works 👌

**Pull request merged**